core/src/main/java/org/mobilitydata/gtfsvalidator/parsing/RowParser.java

```diff
@@ -216,6 +217,18 @@ public String asText(int columnIndex, boolean required) {
 
   @Nullable
   public String asId(int columnIndex, boolean required) {
+    String value = row.asString(columnIndex);
+    if (value == null) {
+      return asString(columnIndex, required);
+    }
+    for (char ch : value.toCharArray()) {
```

Comment:
Maybe a short comment to link to the issue being checked? Doesn't seem that obvious in the context of a RowParser, IMHO